File: 171.excel-sheet-column-number.cpp
/*
 * @lc app=leetcode id=171 lang=cpp
 *
 * [171] Excel Sheet Column Number
 */

// @lc code=start
class Solution {
public:
    int titleToNumber(string columnTitle) {
        int n = columnTitle.length();
        int value = 0;
        int factor = 1;
        
        for (int i = n - 1; i >= 0; i--) {
            char c = columnTitle[i];
            value += (c - 'A' + 1) * factor;
            if (i > 0)
            {
                factor *= 26;
            }
        }
        
        return value;
    }
};

// @lc code=end

File: 171.excel-sheet-column-number.py
#
# @lc app=leetcode id=171 lang=python3
#
# [171] Excel Sheet Column Number
#

# @lc code=start
class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
      n = len(columnTitle)
      value = 0
      for i in range(n):
        c = columnTitle[i]
        value += (ord(c)-ord("A") + 1)*pow(26, n-1-i)
      return value
        
# @lc code=end

